### PR TITLE
Overhaul the source generation docs

### PR DESCRIPTION
**Pull request by cb372**
* Update the sbt plugin name in the docs (sbt-mu-idlgen -> sbt-mu-srcgen)
* Remove any references to IDL generation from the docs, as that is going away soon
* Update the quickstart page to take account of OpenAPI. Rewrite the introduction and remove some unnecessary detail.
* Add a page on OpenAPI source generation
* Update/extend the existing source generation docs
    * The "Generate sources from IDL" page
    * The Avro and Protobuf pages
* Add a section to explain what the `@service` macro annotation is doing

I've published the site to my fork to make it easier to review: https://cb372.github.io/mu-scala/

### CODE REVIEW

**Comment by cb372** [comment]
Note: the failing build is just codecov getting confused.

File: docs/src/main/tut/index.md [hunk 1]
```diff
@@ -8,31 +8,49 @@ permalink: /
...
+* serialization of requests and responses into Avro/Protobuf/JSON
+* handling of streaming requests and responses using either [FS2] or [Monix]
+  Observables.
+* schema management and schema evolution
+* metrics reporting
+* ... and plenty more features on the way!
+
+Specifically, Mu helps you to build:
+
+* [gRPC] servers and clients based on either [Avro] or [Protobuf] protocol definitions
```
Comment (BenFradet) [inline docs/src/main/tut/index.md]:
```diff
-- gRPC
++ RPC
```

Comment (cb372) [reply docs/src/main/tut/index.md]: Really? Isn't it reasonable to say `gRPC` here?

File: docs/src/main/tut/index.md [hunk 2]
```diff
@@ -8,31 +8,49 @@ permalink: /
...
-The current version for [Mu] is "0.20.1" using the following common libraries and versions.
+The current version of [Mu] is "0.20.1".
 
 [comment]: # (End Replace)
 
- * [avro4s] 1.8.4
- * [cats-effect] 2.x
- * [fs2] 2.x
- * [Monix] 3.x
+Mu is cross-built for Scala `2.11.x` and `2.12.x`.
```
[the remark this answers is not in the document]
Comment (cb372) [reply docs/src/main/tut/index.md]: Ok, I'll update this.

File: docs/src/main/tut/index.md (hunk 1 above)
Comment (BenFradet) [reply docs/src/main/tut/index.md]: afaicr Avro uses its own RPC

Comment (cb372) [reply docs/src/main/tut/index.md]: In Mu, they both use gRPC. Here is the code to turn an avro4s codec into a gRPC marshaller: https://github.com/higherkindness/mu-scala/blob/81583cfb2787a6372e3e973ee8708e492e281fb7/modules/internal/src/main/scala/higherkindness/mu/rpc/internal/encoders/avro.scala#L49

Comment (BenFradet) [reply docs/src/main/tut/index.md]: didn't know about this in mu, thanks :+1:

**Pull request merged**